sys/drv/bcm2835_gpio.c

```diff
@@ -71,3 +81,111 @@ void bcm2835_gpio_set_high_detect(resource_t *r, unsigned pin, bool enable) {
     val &= ~mask;
   bus_write_4(r, BCM2835_GPIO_GPHEN(reg), val);
 }
+
+static int bcm2835_gpio_read_fdt_entry(device_t *dev, phandle_t node) {
+  int result = 0;
+  uint32_t *pin_cfgs, *function_cfgs, *pull_cfgs, *intr_detect_cfgs;
+  bcm2835_gpio_t *gpio = (bcm2835_gpio_t *)dev->state;
+
+  ssize_t pin_cnt = 0;
+  ssize_t function_cnt = 0;
+  ssize_t pull_cnt = 0;
+  ssize_t intr_detect_cnt = 0;
+
+  pin_cnt = FDT_getencprop_alloc_multi(node, "pins", sizeof(*pin_cfgs),
+                                       (void **)&pin_cfgs);
+  if (pin_cnt == FDT_PINS_INVAL) {
+    klog("Warning: GPIO FDT entry with no `pins` property");
```

Comment:
@MichalBlk During online interview we decided to add extra messages to `bcm2835_gpio_attach` function so that user knows which set of pins were misconfigured in DTS.